packages/runtime-core/src/componentOptions.ts

```diff
@@ -348,8 +364,24 @@ interface LegacyOptions<
   // since that leads to some sort of circular inference and breaks ThisType
   // for the entire component.
   data?: (
-    this: CreateComponentPublicInstance<Props>,
-    vm: CreateComponentPublicInstance<Props>
+    this: CreateComponentPublicInstance<
+      Props,
+      {},
+      {},
+      ComputedOptions,
```

Comment:
I don't think `computed` are available in `data` since they are initialized after `data`.

Reply:
That's correct, just noticed that the mixin computed will also be inferred which is not correct, but is a bit tricky to extract all computed from nested mixin and extend